Tests/GitHubContents.tests.ps1

```diff
@@ -226,6 +226,121 @@ Describe 'Getting file and folder content' {
             $readmeFileObject.path | Should -Be $readmeFileName
         }
 
+        Context 'When setting new file content by path' {
+            BeforeAll {
+                $filePath = 'notes'
+                $fileName = 'hello.txt'
+                $commitMessage = 'Commit Message'
+                $content = 'This is the content for test.txt'
+                $branchName = 'master'
+                $committerName = 'John Doe'
+                $committerEmail = 'john.doe@testdomain.com'
+                $authorName = 'Jane Doe'
+                $authorEmail = 'jane.doe@testdomain.com'
+
+                $contentPath = New-TemporaryFile
+                $content | Out-File -FilePath $contentPath
+
+                $setGitHubContentParms = @{
+                    Path = "$filePath/$fileName"
+                    CommitMessage = $commitMessage
+                    Branch = $branchName
+                    ContentPath = $contentPath.FullName
+                    Uri = $repo.svn_url
+                    CommitterName = $committerName
+                    CommitterEmail = $committerEmail
+                    authorName = $authorName
+                    authorEmail = $authorEmail
+                }
+
+                $result = Set-GitHubContent @setGitHubContentParms -PassThru
+            }
+
+            It 'Should have the expected type and additional properties' {
+                $result.PSObject.TypeNames[0] | Should -Be 'GitHub.Content'
+                $result.content.name | Should -Be $fileName
+                $result.content.path | Should -Be "$filePath/$fileName"
+                $result.content.url | Should -Be ("https://api.github.com/repos/$($script:ownerName)" +
+                    "/$repoName/contents/$filePath/$($fileName)?ref=$BranchName")
```

Comment:
`$repoName` doesn't exist in this context.  In short, you put these tests in the wrote `Describe` block. They should be part of the block starting at 393 (`Describe 'GitHubContents/Set-GitHubContent' {`)

Reply:
Block moved.